arrow-core-data/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -363,7 +365,11 @@ import arrow.typeclasses.Show
  * Contents partially adapted from [Scala Exercises Option Tutorial](https://www.scala-exercises.org/std_lib/options)
  * Originally based on the Scala Koans.
  */
-
+@Deprecated(
+  "Option will be deleted soon as it promotes the wrong message of using a slower and memory unfriendly " +
+    "abstraction when the lang provides a better one. Alternatively, if you can't support nulls, consider aliasing Either<Unit, A>",
```

Comment:
Can we link to the issue that proposes the alternative and explains 2 use-cases? That's what as an Arrow user I'd go looking for immediately when coming across this.